src/Microsoft.DotNet.Build.VstsBuildsApi/VstsBuildHttpClient.cs

```diff
@@ -19,8 +20,17 @@ public VstsBuildHttpClient(JObject definition, VstsApiEndpointConfig config)
 
         protected override bool IsMatching(JObject localDefinition, JObject retrievedDefinition)
         {
+            string localDefinitionPath = localDefinition["path"].ToString();
+
+            // If it's not the root directory.
+            if (Path.GetDirectoryName(localDefinitionPath) != null) {
```

Comment:
I can't find VSTS docs for the `path` parameter, but I'm not comfortable using `System.IO.Path` to do this since it's a server path rather than on a file system. In my experience it's safe to assume the separator will always be `\`, and I don't know what the behavior is with `/`. Really wish the API docs were complete.

\<nit>Also, the brace should be on the next line.\</nit>